interface/patient_file/summary/demographics.php

```diff
@@ -173,503 +175,517 @@ function image_widget($doc_id, $doc_catg)
-        e.preventDefault();e.stopPropagation();
-        dlgopen('', '', 1000, 600, '', '', {
+        dlgopen(url, 'editScripts', w, 300, '', '', {
             buttons: [
-                {text: <?php echo xlj('Close'); ?>, close: true, style: 'default btn-sm'}
+                {text: <?php echo xlj('Add'); ?>, close: false, id: 'addButton', class: 'btn-primary btn-sm', click: AddScript},
+                {text: <?php echo xlj('Clear'); ?>, close: false, id: 'clearButton', style: 'display:none;', class: 'btn-primary btn-sm', click: AddScript},
+                {text: <?php echo xlj('Back'); ?>, close: false, id: 'backButton', style: 'display:none;', class: 'btn-primary btn-sm', click: ListScripts},
+                {text: <?php echo xlj('Done'); ?>, close: true, id: 'doneButton', class: 'btn-secondary btn-sm'}
             ],
+            onClosed: 'refreshme',
             allowResize: true,
             allowDrag: true,
-            dialogId: '',
-            type: 'iframe',
-            url: $(this).attr('href')
+            dialogId: 'editscripts',
+            type: 'iframe'
         });
-    });
+    }
 
-    $(".rx_modal").on('click', function(e) {
-        e.preventDefault();e.stopPropagation();
-        var AddAmendment = function () {
-            var iam = top.frames.editAmendments;
-            iam.location.href = "<?php echo $GLOBALS['webroot']?>/interface/patient_file/summary/add_edit_amendments.php"
-        };
-        var ListAmendments = function () {
-            var iam = top.frames.editAmendments;
-            iam.location.href = "<?php echo $GLOBALS['webroot']?>/interface/patient_file/summary/list_amendments.php"
-        };
-        var title = <?php echo xlj('Amendments'); ?>;
-        dlgopen('', 'editAmendments', 800, 300, '', title, {
+    function doPublish() {
+        let title = <?php echo xlj('Publish Patient to FHIR Server'); ?>;
+        let url = top.webroot_url + '/phpfhir/providerPublishUI.php?patient_id=' + <?php echo js_url($pid); ?>;
+
+        dlgopen(url, 'publish', 'modal-mlg', 750, '', '', {
             buttons: [
-                {text: <?php echo xlj('Add'); ?>, close: false, style: 'primary  btn-sm', click: AddAmendment},
-                {text: <?php echo xlj('List'); ?>, close: false, style: 'primary  btn-sm', click: ListAmendments},
                 {text: <?php echo xlj('Done'); ?>, close: true, style: 'default btn-sm'}
             ],
-            onClosed: 'refreshme',
             allowResize: true,
             allowDrag: true,
             dialogId: '',
-            type: 'iframe',
-            url: $(this).attr('href')
+            type: 'iframe'
+        });
+    }
+
+    /**
+     * async function fetchHtml(...)
+     *
+     * @param {*} url
+     * @param {boolean} embedded
+     * @param {boolean} sessionRestore
+     * @returns {text}
+     */
+    async function fetchHtml(url, embedded = false, sessionRestore = false) {
+        if (sessionRestore === true) {
```

Comment:
Why is this an option? If this is not done for every html request, then risk making request to the wrong session on the server (in the case when more than 2 tabs are open in browser).
For example, if while these are going, the other tab does a timer script call.

Reply:
Although I see that the original script was doing the same thing since was loading them all in same place, so go figure :)